gensim/models/tfidfmodel.py

```diff
@@ -77,11 +95,69 @@ def __init__(self, corpus=None, id2word=None, dictionary=None,
         If `dictionary` is specified, it must be a `corpora.Dictionary` object
         and it will be used to directly construct the inverse document frequency
         mapping (then `corpus`, if specified, is ignored).
+
+        `smartirs` or SMART (System for the Mechanical Analysis and Retrieval of Text)
+        Information Retrieval System, a mnemonic scheme for denoting tf-idf weighting
+        variants in the vector space model. The mnemonic for representing a combination
+        of weights takes the form ddd, where the letters represents the term weighting
+        of the document vector.
+
+        Term frequency weighing:
+           natural - `n`, logarithm - `l` , augmented - `a`,  boolean `b`, log average - `L`.
+        Document frequency weighting:
+           none - `n`, idf - `t`, prob idf - `p`.
+        Document normalization:
+           none - `n`, cosine - `c`, byte size - `b`.
+
+        for more information visit https://en.wikipedia.org/wiki/SMART_Information_Retrieval_System
         """
-        self.normalize = normalize
+
         self.id2word = id2word
-        self.wlocal, self.wglobal = wlocal, wglobal
+        self.wlocal, self.wglobal, self.normalize = wlocal, wglobal, normalize
         self.num_docs, self.num_nnz, self.idfs = None, None, None
+        self.smartirs = smartirs
+
+        if self.normalize is True:
```

Comment:
What happen if `self.normalize` isn't bool?

Reply:
In that case user if  `smartirs` is not `None`, `self.normalize` will be defined by the values of `smartirs`. If `smartirs` is `None` and  `self.normalize` is not a bool then `self.normalize` must be a function. Just like `wlocal` and `wglobal`.